pkg/apis/core/v1alpha1/federatedtypeconfig_types.go

```diff
@@ -78,7 +79,17 @@ type FederatedTypeConfigStatus struct {
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// FederatedTypeConfig
+// FederatedTypeConfig programs federation to know about a single API resource.
+// For each type, there is a set of API resources that capture the information
+// to federate the type:
+//
+// - A "template" type specifies the basic definition of a resource to be
+//   federated
+// - A "placement" type specifies the placement information for the federated
+//   resource
+// - (optional) A "override" type specifies how parts of a resource are
```

Comment:
Consider using 'the resource should vary across clusters' in place of 'parts of...'